skimage/feature/tests/test_peak.py

```diff
@@ -85,6 +85,56 @@ def test_num_peaks():
     assert (3, 5) in peaks_limited
 
 
+def test_num_peaks_and_labels():
+    image = np.zeros((7, 7), dtype=np.uint8)
+    labels = np.zeros((7, 7), dtype=np.uint8) + 20
+    image[1, 1] = 10
+    image[1, 3] = 11
+    image[1, 5] = 12
+    image[3, 5] = 8
+    image[5, 3] = 7
+    peaks_limited = peak.peak_local_max(
+        image, min_distance=1, threshold_abs=0, labels=labels)
+    assert len(peaks_limited) == 5
+    peaks_limited = peak.peak_local_max(
+        image, min_distance=1, threshold_abs=0, labels=labels, num_peaks=2)
+    assert len(peaks_limited) == 2
+
+def test_num_peaks_per_labels():
+    image = np.zeros((7, 7), dtype=np.uint8)
+    labels = np.zeros((7, 7), dtype=np.uint8) + 1 #0*6.5
+    image[1, 1] = 10
+    image[1, 3] = 11
+    image[1, 5] = 12
+    image[3, 5] = 8
+    image[5, 3] = 7
+    peaks_limited = peak.peak_local_max(
+        image, min_distance=1, threshold_abs=0, labels=labels)
+    assert len(peaks_limited) == 5
+    peaks_limited = peak.peak_local_max(
+        image, min_distance=1, threshold_abs=0, labels=labels, num_peaks_per_label=2)
+    assert len(peaks_limited) == 2
+
+
+def test_num_peaks_and_labels_4quadrants():
+    image = np.random.uniform(size=(40, 60))
```

Comment:
Please, specify the seed. We may face assertion fail from time to time, if `/dev/random` has bad luck.

The test itself is OK, but I'd consider decreasing the image size. Let's say to 20x20.